File: enclose/tests/fixtures/sample.c
#include <stdio.h>

struct widget {
    int size;
};

static int helper(int value)
{
    return value + 1;
}

int grow(struct widget *w)
{
    w->size = helper(w->size);
    return w->size;
}

int main(void)
{
    struct widget w = { 3 };
    printf("%d\n", grow(&w));
    return 0;
}

File: enclose/tests/fixtures/sample.rs
struct Widget {
    size: i32,
}

fn helper(value: i32) -> i32 {
    value + 1
}

impl Widget {
    fn grow(&mut self) -> i32 {
        self.size = helper(self.size);
        self.size
    }
}

fn main() {
    let mut w = Widget { size: 3 };
    let bump = |x: i32| helper(x);
    println!("{}", bump(w.grow()));
}
